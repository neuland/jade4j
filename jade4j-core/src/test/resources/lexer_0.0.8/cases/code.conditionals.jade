
- if (true)
  p foo
- else
  p bar

- if (true) {
  p foo
- } else {
  p bar
- }

if true
  p foo
  p bar
  p baz
else
  p bar

unless true
  p foo
else
  p bar

if 'nested'
  if 'works'
    p yay

//- allow empty blocks
if false
else
  .bar
if true
  .bar
else
.bing