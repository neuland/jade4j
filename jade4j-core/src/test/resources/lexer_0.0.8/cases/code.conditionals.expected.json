{"type":"newline","line":2}
{"type":"code","line":2,"val":"if (true)","escape":false,"buffer":false}
{"type":"indent","line":3,"val":2}
{"type":"tag","line":3,"val":"p","selfClosing":false}
{"type":"text","line":3,"val":"foo"}
{"type":"outdent","line":4}
{"type":"code","line":4,"val":"else","escape":false,"buffer":false}
{"type":"indent","line":5,"val":2}
{"type":"tag","line":5,"val":"p","selfClosing":false}
{"type":"text","line":5,"val":"bar"}
{"type":"outdent","line":7}
{"type":"code","line":7,"val":"if (true) {","escape":false,"buffer":false}
{"type":"indent","line":8,"val":2}
{"type":"tag","line":8,"val":"p","selfClosing":false}
{"type":"text","line":8,"val":"foo"}
{"type":"outdent","line":9}
{"type":"code","line":9,"val":"} else {","escape":false,"buffer":false}
{"type":"indent","line":10,"val":2}
{"type":"tag","line":10,"val":"p","selfClosing":false}
{"type":"text","line":10,"val":"bar"}
{"type":"outdent","line":11}
{"type":"code","line":11,"val":"}","escape":false,"buffer":false}
{"type":"newline","line":13}
{"type":"code","line":13,"val":"if ( true)","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":14,"val":2}
{"type":"tag","line":14,"val":"p","selfClosing":false}
{"type":"text","line":14,"val":"foo"}
{"type":"newline","line":15}
{"type":"tag","line":15,"val":"p","selfClosing":false}
{"type":"text","line":15,"val":"bar"}
{"type":"newline","line":16}
{"type":"tag","line":16,"val":"p","selfClosing":false}
{"type":"text","line":16,"val":"baz"}
{"type":"outdent","line":17}
{"type":"code","line":17,"val":"else","isElse":true,"isIf":false,"requiresBlock":true}
{"type":"indent","line":18,"val":2}
{"type":"tag","line":18,"val":"p","selfClosing":false}
{"type":"text","line":18,"val":"bar"}
{"type":"outdent","line":20}
{"type":"code","line":20,"val":"if (!( true))","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":21,"val":2}
{"type":"tag","line":21,"val":"p","selfClosing":false}
{"type":"text","line":21,"val":"foo"}
{"type":"outdent","line":22}
{"type":"code","line":22,"val":"else","isElse":true,"isIf":false,"requiresBlock":true}
{"type":"indent","line":23,"val":2}
{"type":"tag","line":23,"val":"p","selfClosing":false}
{"type":"text","line":23,"val":"bar"}
{"type":"outdent","line":25}
{"type":"code","line":25,"val":"if ( 'nested')","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":26,"val":2}
{"type":"code","line":26,"val":"if ( 'works')","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":27,"val":4}
{"type":"tag","line":27,"val":"p","selfClosing":false}
{"type":"text","line":27,"val":"yay"}
{"type":"outdent","line":29}
{"type":"outdent","line":29}
{"type":"comment","line":29,"val":" allow empty blocks","buffer":false}
{"type":"newline","line":30}
{"type":"code","line":30,"val":"if ( false)","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"newline","line":31}
{"type":"code","line":31,"val":"else","isElse":true,"isIf":false,"requiresBlock":true}
{"type":"indent","line":32,"val":2}
{"type":"class","line":32,"val":"bar"}
{"type":"outdent","line":33}
{"type":"code","line":33,"val":"if ( true)","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":34,"val":2}
{"type":"class","line":34,"val":"bar"}
{"type":"outdent","line":35}
{"type":"code","line":35,"val":"else","isElse":true,"isIf":false,"requiresBlock":true}
{"type":"newline","line":36}
{"type":"class","line":36,"val":"bing"}
{"type":"eos","line":36}